:book: Add Clusterclass and RuntimeSDK terms to glossary (#6845)
kubernetes-sigs/cluster-api: docs/book/src/reference/glossary.md

A feature implementation offered as part of the Cluster API project, infrastructure providers can swap it out for a different one.

# E
---

### External patch

[Patch](#patch) generated by an external component using [Runtime SDK](#runtime-sdk).

### External patch extension

> An external component leveraging [Runtime SDK](#runtime-sdk) that generates [patches](#patch) .
> 
> See [Runtime SDK](#runtime-sdk)

Reviewer: ```suggestion
A [runtime extension] that implements a [topology mutation hook]

See [runtime extension](#runtime-extension), [topology mutation hook](#topology-mutation-hook)
```
wdyt?

A source of computational resources (e.g. machines, networking, etc.). Examples for cloud include AWS, Azure, Google, etc.; for bare metal include VMware, MAAS, metal3.io, etc. When there is more than one way to obtain resources from the same infrastructure provider (e.g. EC2 vs. EKS) each way is referred to as a variant.

### Inline patch 

> A [patch](#patch) defined inline in a [ClusterClass](#clusterclass) and implemented by the core CAPI controller.

Reviewer: ```suggestion
A [patch](#patch) defined inline in a [ClusterClass](#clusterclass) using JSON patch syntax.
```

# P
---

> ### Patch 

Author: This is one that wasn't defined in a proposal, but seemed like it needed a definition here. @sbueringer WDYT?

Reviewer 2: No objections

# P
---

### Patch 

> A well-defined set of interfaces and APIs to describe modifications to a Kubernetes object.

Reviewer: ```suggestion
A set of instructions describing modifications to a Kubernetes object. Depending on the context it could be either a JSON Patch or a JSON Merge Patch.
```
wdyt?

Author: I'd prefer to leave this high level and not to go over and back too on the wording @vincepri WDYT?

Reviewer: TBH I have some concerns given that I struggle to identify a patch as a "well-defined set of interfaces and APIs"

Existing Cluster API implementations consist of generic and infrastructure provider-specific logic. The [infrastructure provider](#infrastructure-provider)-specific logic is currently maintained in infrastructure provider repositories.

# R
---

### Runtime Extension

> An external component which is part of a system built on top of Cluster API that can handle requests for a specific Runtime Hook.

Author: I think handlers are more clearly on the implementation side and we should limit this doc to high-level vocabulary rather than trying to re-define out API types here.

The infrastructure that backs a [Machine Resource](#machine), typically either a cloud instance, virtual machine, or physical host.

# T
---

### Topology

> A topology is a set of fields in a Cluster object that define the cluster.

Reviewer: ```suggestion
A topology is a struct in the Cluster object spec that allows to define and manage the shape cluster, control plane and workers machines, from a single point of control. The cluster's topology is based on a [ClusterClass](#clusterclass).
```
wdyt?
wdyt?

Author: I'd prefer not to go too deep in the specifics here as I think that keeps us in line with the rest of the glossary - e.g. the current definition for Cluster is `A full Kubernetes deployment.`  If we want to revise the overall glossary let's put this on hold and decide goals of this page are, or I can open a separate issue for that and we can do it in parallel?

Reviewer: TBH I'm not concerned about adding some more details. In this case, we are still using a single sentence and IMO it gives some useful context

See [ClusterClass](#clusterclass)

### Topology Mutation Hook

> A [Runtime Hook](#runtime-hook) that follows well-defined interfaces to generate modifications to Kubernetes objects.

Reviewer: ```suggestion
A [Runtime Hook](#runtime-hook) that allows external components to generate [patches](#patch) for customizing the Kubernetes objects that implements a [Cluster topology](#topology).
```
